scripts/ipfs/ipfs-setup.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/bash
+
+# This script connects to local mesh peers,
+# and it sets up connection filters based on what networks this node can access.
+# It runs continually, to change IPFS settings as the environment around the node changes.
+
+
+# Wait for IPFS to initalize
+attempts=15
+until [[ $(curl http://localhost:5001/api/v0/id -s 2>/dev/null) || ${attempts} -eq 0 ]]; do
+    sleep 1
+    attempts=$((attempts-1))
+done
+
+if [[  ${attempts} -eq 0 ]]; then
+    echo "Error: Failed to connect to local IPFS daemon. Is it running?"
+    exit 1
+fi
+
+# Update peers's data since IPFS just started
+echo "Updating nodeinfo..."
+sudo /usr/local/bin/nodeinfo-update.sh
+
+function addPeer  {
+    addr=$1
+    # See if they have IPFS enabled
+    res=$(curl http://["${addr}"]/nodeinfo.json -s)
+    if [ ! -x "${res}" ]; then
+        id=$(echo "${res}" | jq -r -M '.services.ipfs.ID')
+        # Value is found
+        if [[ ! ${id} == "null" ]] && [[ ! "${id}" == "" ]]; then
+            # Connect to neighbouring IPFS nodes
+            # Check for QUIC connections first
+            if [ "$(echo "${res}" | jq -r -M '.services.IPFS.quic_enabled')" == 'true' ]; then
+                # ID is not needed for QUIC connections
+                ipfs swarm connect "/ip6/${addr}/udp/4001/quic"
+                echo "Connecting to ${addr} with QUIC"
+            else
+                ipfs swarm connect "/ip6/${addr}/tcp/4001/ipfs/${id}"
+                echo "Connecting to ${addr} regularly"
+            fi
+        fi
+    fi
+}
+
+# Continually mesh with peers and add or remove filter rules as networks go up and down
+while true; do
```

Comment:
I dont know how `postexec` will work for long running processes. Upon service restart I dont think the PID will be killed but it will spawn again.

This will created multiple scripts running in parallel.

Better off running in a service